### PR TITLE
Raise exceptions to track in Sentry

### PR DESCRIPTION
**Pull request by armenzg**
This fixes reporting of autoland cloning failures.

### CODE REVIEW

File: http_service/bugbug_http/models.py [hunk 1]
```diff
@@ -121,15 +121,15 @@ def schedule_tests(branch: str, rev: str) -> str:
         revs = get_hgmo_stack(branch, rev)
     except requests.exceptions.RequestException:
         LOGGER.warning(f"Push not found for {branch} @ {rev}!")
-        return "NOK"
+        raise
```
Comment (marco-c) [inline http_service/bugbug_http/models.py:124]: In this case we should keep returning "NOK", if the push can't be found it's not useful to retry

File: http_service/bugbug_http/boot.py [hunk 2]
```diff
@@ -97,10 +97,10 @@ def extract_commit_experiences() -> None:
             utils.extract_file(os.path.join("data", repository.COMMIT_EXPERIENCES_DB))
             logger.info("Commit experiences DB extracted.")
         except FileNotFoundError:
+            assert ALLOW_MISSING_MODELS
```
Comment (marco-c) [inline http_service/bugbug_http/boot.py:100]: Can you make this change in a separate PR?

File: http_service/bugbug_http/models.py (hunk 1 above)
Comment (armenzg) [reply http_service/bugbug_http/models.py]: This was left unstaged. Fixed now.

File: http_service/bugbug_http/boot.py (hunk 2 above)
Comment (armenzg) [reply http_service/bugbug_http/boot.py]: Done https://github.com/mozilla/bugbug/pull/1720
